From c9670b5a63ac14ad239610bde30ee61b2deee58d Mon Sep 17 00:00:00 2001
From: Howard Pritchard <hppritcha@gmail.com>
Date: Thu, 20 Dec 2018 14:36:53 -0700
Subject: [PATCH] don't try to open HFI devices with UCX

Signed-off-by: Howard Pritchard <hppritcha@gmail.com>

diff --git a/src/uct/ib/base/ib_device.c b/src/uct/ib/base/ib_device.c
index feab36be..f9f0a06d 100644
--- a/src/uct/ib/base/ib_device.c
+++ b/src/uct/ib/base/ib_device.c
@@ -249,6 +249,13 @@ ucs_status_t uct_ib_device_init(uct_ib_device_t *dev,
         goto err_free_context;
     }
 
+    if (strstr(ibv_get_device_name(ibv_device), "hfi") != NULL) {
+        ucs_debug("skipping Intel OPA100 device %s since not supported by UCX",
+                  ibv_get_device_name(ibv_device));
+        status = UCS_ERR_IO_ERROR;
+        goto err_free_context;
+    }
+
     /* Check device type*/
     switch (ibv_device->node_type) {
     case IBV_NODE_SWITCH:
-- 
1.8.3.1

