diff --git a/ompi/mca/pml/ucx/pml_ucx_component.c b/ompi/mca/pml/ucx/pml_ucx_component.c
index ff0040f..a0be432 100644
--- a/ompi/mca/pml/ucx/pml_ucx_component.c
+++ b/ompi/mca/pml/ucx/pml_ucx_component.c
@@ -8,12 +8,13 @@
  */
 
 #include "pml_ucx.h"
-
 #include "opal/memoryhooks/memory.h"
 #include "opal/mca/memory/base/base.h"
 
 #include <ucm/api/ucm.h>
-
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
 
 static int mca_pml_ucx_component_register(void);
 static int mca_pml_ucx_component_open(void);
@@ -130,7 +131,10 @@ mca_pml_ucx_component_init(int* priority, bool enable_progress_threads,
                              bool enable_mpi_threads)
 {
     int ret;
-
+    if ((0 == access("/sys/class/ugni/", F_OK) || (0 == access("/sys/class/hfi1/", F_OK)))){
+	PML_UCX_VERBOSE(1, "Cray or Intel HSN detected, removing UCX from consideration");
+	return NULL;
+    } 
     if ( (ret = mca_pml_ucx_init()) != 0) {
         return NULL;
     }
