diff -ur charliecloud/bin/ch-build charliecloud-lib/bin/ch-build
--- charliecloud/bin/ch-build	2020-04-01 14:59:49.394882226 -0600
+++ charliecloud-lib/bin/ch-build	2020-04-15 15:50:09.377151707 -0600
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -ur charliecloud/bin/ch-build2dir charliecloud-lib/bin/ch-build2dir
--- charliecloud/bin/ch-build2dir	2020-04-01 14:59:49.394882226 -0600
+++ charliecloud-lib/bin/ch-build2dir	2020-04-15 15:50:19.707251954 -0600
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -ur charliecloud/bin/ch-builder2squash charliecloud-lib/bin/ch-builder2squash
--- charliecloud/bin/ch-builder2squash	2020-04-01 14:59:49.394882226 -0600
+++ charliecloud-lib/bin/ch-builder2squash	2020-04-15 15:50:39.278441877 -0600
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -ur charliecloud/bin/ch-builder2tar charliecloud-lib/bin/ch-builder2tar
--- charliecloud/bin/ch-builder2tar	2020-04-01 14:59:49.394882226 -0600
+++ charliecloud-lib/bin/ch-builder2tar	2020-04-15 15:50:54.034585075 -0600
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -ur charliecloud/bin/ch-dir2squash charliecloud-lib/bin/ch-dir2squash
--- charliecloud/bin/ch-dir2squash	2020-04-01 14:59:49.394882226 -0600
+++ charliecloud-lib/bin/ch-dir2squash	2020-04-15 15:51:08.922729551 -0600
@@ -2,7 +2,7 @@
 
 set -e
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -ur charliecloud/bin/ch-fromhost charliecloud-lib/bin/ch-fromhost
--- charliecloud/bin/ch-fromhost	2020-04-01 14:59:49.395882236 -0600
+++ charliecloud-lib/bin/ch-fromhost	2020-04-15 15:51:29.832932469 -0600
@@ -35,7 +35,7 @@
 #   3. Here string, e.g. 'while IFS= read -r FILE; do ... done <<< "$FILES"'.
 #      This is a bashism.
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 set -e
diff -ur charliecloud/bin/ch-grow.py.in charliecloud-lib/bin/ch-grow.py.in
--- charliecloud/bin/ch-grow.py.in	2020-04-07 12:19:37.053609696 -0600
+++ charliecloud-lib/bin/ch-grow.py.in	2020-04-15 15:51:46.414093375 -0600
@@ -17,7 +17,7 @@
 import types
 
 sys.path.insert(0, (  os.path.dirname(os.path.abspath(__file__))
-                    + "/../lib/charliecloud"))
+                    + "/../lib64/charliecloud"))
 import charliecloud as ch
 
 
diff -ur charliecloud/bin/ch-mount charliecloud-lib/bin/ch-mount
--- charliecloud/bin/ch-mount	2020-04-01 14:59:49.395882236 -0600
+++ charliecloud-lib/bin/ch-mount	2020-04-15 15:52:08.540308096 -0600
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -ur charliecloud/bin/ch-pull2dir charliecloud-lib/bin/ch-pull2dir
--- charliecloud/bin/ch-pull2dir	2020-04-01 14:59:49.395882236 -0600
+++ charliecloud-lib/bin/ch-pull2dir	2020-04-15 15:52:18.332403120 -0600
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 set -e
diff -ur charliecloud/bin/ch-pull2tar charliecloud-lib/bin/ch-pull2tar
--- charliecloud/bin/ch-pull2tar	2020-04-01 14:59:49.395882236 -0600
+++ charliecloud-lib/bin/ch-pull2tar	2020-04-15 15:52:32.234538029 -0600
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 set -e
diff -ur charliecloud/bin/ch-run-oci.py.in charliecloud-lib/bin/ch-run-oci.py.in
--- charliecloud/bin/ch-run-oci.py.in	2020-04-01 14:59:49.395882236 -0600
+++ charliecloud-lib/bin/ch-run-oci.py.in	2020-04-15 15:52:45.887670523 -0600
@@ -14,7 +14,7 @@
 import types
 
 sys.path.insert(0, (  os.path.dirname(os.path.abspath(__file__))
-                    + "/../lib/charliecloud"))
+                    + "/../lib64/charliecloud"))
 import version
 
 
diff -ur charliecloud/bin/ch-tar2dir charliecloud-lib/bin/ch-tar2dir
--- charliecloud/bin/ch-tar2dir	2020-04-01 14:59:49.396882245 -0600
+++ charliecloud-lib/bin/ch-tar2dir	2020-04-15 15:53:09.747902070 -0600
@@ -2,7 +2,7 @@
 
 set -e
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
diff -ur charliecloud/bin/ch-test charliecloud-lib/bin/ch-test
--- charliecloud/bin/ch-test	2020-04-07 12:19:37.054609706 -0600
+++ charliecloud-lib/bin/ch-test	2020-04-15 15:53:21.830019318 -0600
@@ -20,7 +20,7 @@
 
 ### Setup
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 if [[ ! -f ${lib}/base.sh ]]; then
     fatal "install or build problem: not found: ${lib}/base.sh"
 fi
diff -ur charliecloud/bin/ch-tug.py.in charliecloud-lib/bin/ch-tug.py.in
--- charliecloud/bin/ch-tug.py.in	2020-04-01 14:59:49.396882245 -0600
+++ charliecloud-lib/bin/ch-tug.py.in	2020-04-15 15:53:39.030186232 -0600
@@ -5,7 +5,7 @@
 import sys
 
 sys.path.insert(0, (  os.path.dirname(os.path.abspath(__file__))
-                    + "/../lib/charliecloud"))
+                    + "/../lib64/charliecloud"))
 import charliecloud as ch
 
 
diff -ur charliecloud/bin/ch-umount charliecloud-lib/bin/ch-umount
--- charliecloud/bin/ch-umount	2020-04-07 12:19:37.054609706 -0600
+++ charliecloud-lib/bin/ch-umount	2020-04-15 15:53:52.979321598 -0600
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-lib=$(cd "$(dirname "$0")" && pwd)/../lib/charliecloud
+lib=$(cd "$(dirname "$0")" && pwd)/../lib64/charliecloud
 . "${lib}/base.sh"
 
 # shellcheck disable=SC2034
Only in charliecloud/packaging/fedora: ch-test.lib.patch
