diff -ru charliecloud/bin/ch-test charliecloud-lib/bin/ch-test
--- charliecloud/bin/ch-test	2020-04-07 12:19:37.054609706 -0600
+++ charliecloud-lib/bin/ch-test	2020-04-15 16:36:55.128831767 -0600
@@ -662,7 +662,7 @@
     CHTEST_INSTALLED=yes
     CHTEST_GITWD=
     CHTEST_DIR=${ch_base}/libexec/charliecloud/test
-    CHTEST_EXAMPLES_DIR=${ch_base}/share/doc/charliecloud/examples
+    CHTEST_EXAMPLES_DIR=${ch_base}/share/doc/charliecloud-${ch_version}/examples
 else
     # build dir
     CHTEST_INSTALLED=
