--- a/bin/ch-test	2024-01-19 13:17:53.000000000 -0500
+++ b/bin/ch-test	2024-02-09 14:37:15.388753491 -0500
@@ -779,7 +779,7 @@

 # Find test directories. Note some of this gets rewritten at install time.
 CHTEST_DIR=${ch_base}/test
-CHTEST_EXAMPLES_DIR=${ch_base}/examples
+CHTEST_EXAMPLES_DIR=${ch_base}-${ch_version}/examples
 if [[ ! -f ${ch_base}/VERSION ]]; then
     # installed
     CHTEST_INSTALLED=yes
