--- charliecloud-0.32/bin/ch-test    2023-03-22 17:36:46.000000000 -0400
+++ charliecloud-0.32.patch/bin/ch-test    2023-04-03 10:51:53.677205658 -0400
@@ -767,7 +767,7 @@
     CHTEST_INSTALLED=yes
     CHTEST_GITWD=
     CHTEST_DIR=${ch_base}/libexec/charliecloud/test
-    CHTEST_EXAMPLES_DIR=${ch_base}/share/doc/charliecloud/examples
+    CHTEST_EXAMPLES_DIR=${ch_base}/share/doc/charliecloud-${ch_version}/examples
 else
     # build dir
     CHTEST_INSTALLED=
