diff -ur charliecloud/bin/ch_misc.c charliecloud-patch/bin/ch_misc.c
--- charliecloud/bin/ch_misc.c	2022-01-24 13:12:23.980046774 -0500
+++ charliecloud-patch/bin/ch_misc.c	2022-01-24 13:25:34.854133321 -0500
@@ -252,7 +252,7 @@
    if (path == NULL) {
       T_ (where = strdup(line));
    } else {
-      T_ (1 <= asprintf(&where, "%s:%lu", path, lineno));
+      T_ (1 <= asprintf(&where, "%s:%zu", path, lineno));
    }
 
    // Split line into variable name and value.
