--- packaging/fedora/charliecloud.spec	2024-02-12 11:46:23.034342881 -0700
+++ packaging/fedora/upstream.spec	2024-02-09 14:32:41.000000000 -0700
@@ -10,17 +10,15 @@
 %{?el7:%global __python %__python3}
 
 Name:          charliecloud
-Version:       @VERSION@
-Release:       @RELEASE@%{?dist}
+Version:       0.36
+Release:       2%{?dist}
 Summary:       Lightweight user-defined software stacks for high-performance computing
 License:       ASL 2.0
 URL:           https://hpc.github.io/%{name}/
 Source0:       https://github.com/hpc/%{name}/releases/downloads/v%{version}/%{name}-%{version}.tar.gz
 BuildRequires: gcc rsync bash findutils
-%if 0%{?el7}
 Patch0:        el7-pkgdir.patch
-%endif
-%if 0%{!?el7} && 0%{!?el8}
+%if 0%{?fedora} > 36 || 0%{?rhel} > 8
 Requires:      fuse3 squashfuse
 BuildRequires: fuse3-libs fuse3-devel squashfuse-devel
 Patch1:        no-squashfuse-rpath.patch
@@ -45,14 +43,11 @@
 Requires:      %{name}
 Requires:      python3
 Requires:      python%{python3_pkgversion}-requests
-
-%if 0%{?el7}
-Requires:      git
-%else
+%if 1%{?el7}
 Requires:      git >= 2.28.1
 %endif
-
 Provides:      bundled(python%{python3_pkgversion}-lark-parser) = 1.1.9
+%{?el7:BuildArch: noarch}
 
 %description builder
 This package provides ch-image, Charliecloud's completely unprivileged container
@@ -73,7 +68,7 @@
 %package   test
 Summary:   Charliecloud test suite
 License:   ASL 2.0
-Requires:  %{name} %{name}-builder bats
+Requires:  %{name} %{name}-builder /usr/bin/bats
 Obsoletes: %{name}-test < %{version}-%{release}
 
 %description test
@@ -86,7 +81,7 @@
 %patch0 -p1
 %endif
 
-%if 0%{!?el7}
+%if 0%{?fedora} > 36
 %patch 1 -p1
 %endif
 
@@ -94,16 +89,15 @@
 # Use old inlining behavior, see:
 # https://github.com/hpc/charliecloud/issues/735
 CFLAGS=${CFLAGS:-%optflags -fgnu89-inline}; export CFLAGS
-# FIXME: use --disable test when https://github.com/hpc/charliecloud/issues/1836
-# is resolved.
 %configure --docdir=%{_pkgdocdir} \
            --libdir=%{_prefix}/lib \
            --with-python=/usr/bin/python3 \
+%if 0%{?fedora} > 34 || 0%{?rhel} > 8
            --with-libsquashfusei=/usr \
+%endif
 %if 0%{?el7}
            --with-sphinx-build=%{_bindir}/sphinx-build-3.6
 %else
-           --with-libsquashfusei=/usr \
            --with-sphinx-build=%{_bindir}/sphinx-build
 %endif
 
@@ -140,13 +134,6 @@
 %{__rm} -f %{buildroot}%{_pkgdocdir}/LICENSE
 %{__rm} -f %{buildroot}%{_pkgdocdir}/README.rst
 
-# FIXME: unnecessary after #1836 is resolved
-%if 0%{?el7}
-%{__rm} -f  %{buildroot}%{_bindir}/ch-test
-%{__rm} -rf %{buildroot}%{_libexecdir}/%{name}
-%{__rm} -f  %{buildroot}%{_mandir}/man1/ch-test.1*
-%endif
-
 %files
 %license LICENSE
 %doc README.rst %{?el7:README.EL7}
@@ -191,13 +178,224 @@
 %{?el7:%exclude %{_pkgdocdir}/examples/*/__pycache__}
 
 %files test
-%if 0%{?el7}
-%else
 %{_bindir}/ch-test
 %{_libexecdir}/%{name}
 %{_mandir}/man1/ch-test.1*
-%endif
 
 %changelog
-* Thu Apr 16 2020 <jogas@lanl.gov> - @VERSION@-@RELEASE@
-- Add new charliecloud package.
+* Fri Feb 09 2024 Jordan Ogas <jogas@lanl.gov> - 0.36-2
+- fix epel7 patch
+
+* Fri Feb 09 2024 Jordan Ogas <jogas@lanl.gov> - 0.36-1
+- new version 0.36
+
+* Tue Jan 23 2024 Fedora Release Engineering <releng@fedoraproject.org> - 0.32-5
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_40_Mass_Rebuild
+
+* Fri Jan 19 2024 Fedora Release Engineering <releng@fedoraproject.org> - 0.32-4
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_40_Mass_Rebuild
+
+* Wed Jul 19 2023 Fedora Release Engineering <releng@fedoraproject.org> - 0.32-3
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_39_Mass_Rebuild
+
+* Mon Apr 03 2023 Jordan Ogas <jogas@lanl.gov> 0.32-2
+- fix macro conditionals
+
+* Fri Mar 31 2023 Jordan Ogas <jogas@lanl.gov> 0.32-1
+- edit el7 patch: update ch-test path
+- add patch: prevent rpath of standard path
+- update builder package files
+- add dependency: findutils
+- add conditional dependencies: fuse3, squashfuse-devel, & git
+- new version 0.32
+
+* Wed Jan 18 2023 Fedora Release Engineering <releng@fedoraproject.org> - 0.26-3
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_38_Mass_Rebuild
+
+* Wed Jul 20 2022 Fedora Release Engineering <releng@fedoraproject.org> - 0.26-2
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_37_Mass_Rebuild
+
+* Mon Jan 24 2022 Jordan Ogas <jogas@lanl.gov> 0.26-1
+- add printf patch for 32-bit
+- add ch-convert script
+- new version 0.26
+
+* Wed Jan 19 2022 Fedora Release Engineering <releng@fedoraproject.org> - 0.25-2
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_36_Mass_Rebuild
+
+* Mon Sep 20 2021 Jordan Ogas <jogas@lanl.gov> 0.25-1
+- bundle python lark parser
+- new version
+
+* Thu Aug 05 2021 Jordan Ogas <jogas@lanl.gov> 0.24-12
+- remove version numbers from Obsolete
+- remove Provides tag
+- replace package name with macro
+- tidy
+
+* Thu Jul 29 2021 Jordan Ogas <jogas@lanl.gov> 0.24-11
+- move -builder to noarch
+- move examples back to -doc
+- add versions to obsoletes
+- use name macro
+
+* Wed Jul 28 2021 Jordan Ogas <jogas@lanl.gov> 0.24-10
+- fix yet another typo; BuildRequires
+
+* Wed Jul 28 2021 Jordan Ogas <jogas@lanl.gov> 0.24-9
+- add version to obsoletes
+
+* Wed Jul 28 2021 Jordan Ogas <jogas@lanl.gov> 0.24-8
+- fix provides typo
+
+* Wed Jul 28 2021 Jordan Ogas <jogas@lanl.gov> 0.24-7
+- add -common to obsoletes and provides
+
+* Wed Jul 28 2021 Jordan Ogas <jogas@lanl.gov> - 0.24-6
+* revert to meta-package; separate builder to -builder
+
+* Wed Jul 21 2021 Fedora Release Engineering <releng@fedoraproject.org> - 0.24-5
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_35_Mass_Rebuild
+
+* Mon Jul 19 2021 Jordan Ogas <jogas@lanl.gov> - 0.24-4
+- fix epel7 python cache files
+
+* Mon Jul 19 2021 Jordan Ogas <jogas@lanl.gov> - 0.24-3
+- Tidy, alphabatize files
+- Move builder exlusive python files out from -common
+- Move generic helper scripts to -common
+- Add requires runtime to -builders
+
+* Tue Jul 13 2021 Dave Love <loveshack@fedoraproject.org> - 0.24-2
+- Obsolete previous packge by -runtime, not -common
+
+* Wed Jun 30 2021 Dave Love <loveshack@fedoraproject.org> - 0.24-1
+- New version
+
+* Sun Apr 18 2021 Dave Love <loveshack@fedoraproject.org> - 0.23-1
+- New version
+- Split main package into runtime, builder, and common sub-packages
+- Require buildah and squashfs at run time
+- Use /lib, not /lib64 for noarch; drop lib64 patch
+- Don't BR squashfs-tools, squashfuse, buildah
+- Require squashfs-tools in -builders
+
+* Mon Mar 8 2021 Dave Love <loveshack@fedoraproject.org> <jogas@lanl.gov> - 0.22-2
+- Fix source0 path
+- Put man7 in base package
+
+* Tue Feb 9 2021 Dave Love <loveshack@fedoraproject.org> <jogas@lanl.gov> - 0.22-1
+- New version
+- update lib64.patch
+- add pull.py and push.py
+- (Build)Require python3-lark-parser, python3-requests
+
+* Wed Feb 3 2021 <jogas@lanl.gov> - 0.21-2
+- Fix lib64.patch path for ch-image
+
+* Tue Jan 05 2021 <loveshack@fedoraproject.org> <jogas@lanl.gov> - 0.21-1
+- New version
+- Ship charlicloud.7
+- Require fakeroot
+- Install fakeroot.py
+- Always ship patch1
+- Get python3_sitelib defined
+- Move examples to -test and require sphinx_rtd_theme
+- Include __pycache__ on el7
+- Use %%python3_pkgversion
+- BR python3, not /usr/bin/python3
+- Fix comment capitalization and spacing
+
+* Tue Sep 22 2020 <jogas@lanl.gov> - 0.19-1
+- Package build.py and misc.py
+- Remove unnecessary patch
+- New release
+
+* Mon Jul 27 2020 Fedora Release Engineering <releng@fedoraproject.org> - 0.15-2
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_33_Mass_Rebuild
+
+* Thu Apr 16 2020 <jogas@lanl.gov> - 0.15-1
+- Add test suite package
+- Update spec for autoconf
+- New release
+
+* Tue Jan 28 2020 Fedora Release Engineering <releng@fedoraproject.org> - 0.10-2
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_32_Mass_Rebuild
+
+* Wed Sep 04 2019 <jogas@lanl.gov> - 0.10-1
+- Patch doc-src/conf.py for epel
+- Fix doc-src/dev.rst
+- Fix libexec and doc install path for 0.10 changes
+- Tidy comments
+- New release
+
+* Thu Aug 22 2019 <jogas@lanl.gov> - 0.9.10-12
+- Upate doc subpackage obsoletes
+
+* Mon Aug 19 2019 Dave love <loveshack@fedoraproject.org> - 0.9.10-11
+- Use canonical form for Source0
+- Remove main package dependency from doc, and make it noarch
+
+* Fri Aug 02 2019 <jogas@lanl.gov> 0.9.10-10
+- Tidy comments; fix typ
+
+* Thu Jul 25 2019 <jogas@lanl.gov> 0.9.10-9
+- Use python site variable; fix doc file reference
+
+* Tue Jul 23 2019 <jogas@lanl.gov> 0.9.10-8
+- Remove bundled js, css, and font bits
+
+* Mon Jul 22 2019 <jogas@lanl.gov 0.9.10-7
+- Fix prep section to handle warnings
+- Move documentation dependencies to doc package
+
+* Fri Jul 19 2019 <jogas@lanl.gov> 0.9.10-6
+- Temporarily remove test suite
+
+* Wed Jul 10 2019 <jogas@lanl.gov> 0.9.10-5
+- Revert test and example install path change
+- Update test readme
+
+* Wed Jul 3 2019 <jogas@lanl.gov> 0.9.10-4
+- Add doc package
+
+* Tue Jul 2 2019 <jogas@lanl.gov> 0.9.10-3
+- Tidy comments
+- Update source URL
+- Build html documentation; add rsync dependency
+- Add el7 conditionals for documentation
+- Remove libexecdir definition
+- Add test suite README.TEST
+
+* Wed May 15 2019  <jogas@lanl.gov> 0.9.10-2
+- Fix comment typo
+- Move test suite install path
+
+* Tue May 14 2019  <jogas@lanl.gov> 0.9.10-1
+- New version
+- Fix README.EL7 sysctl command instruction
+- Add pre-built html documentation
+- Fix python dependency
+- Remove temporary test-package readme
+- Fixed capitalization of change log messages
+
+* Tue Apr 30 2019  <jogas@lanl.gov> 0.9.9-4
+- Move global python declaration
+
+* Mon Apr 29 2019  <jogas@lanl.gov> 0.9.9-3
+- Match bin files with wildcard
+
+* Mon Apr 29 2019  <jogas@lanl.gov> 0.9.9-2
+- Update macro comment
+- Fix release tag history
+
+* Tue Apr 16 2019  <jogas@lanl.gov> 0.9.9-1
+- New version
+- Move temp readme creation to install segment
+- Fix spec file macro
+
+* Tue Apr 02 2019  <jogas@lanl.gov> 0.9.8-2
+- Remove python2 build option
+
+* Thu Mar 14 2019  <jogas@lanl.gov> 0.9.8-1
+- Add initial Fedora/EPEL package
